### PR TITLE
New maintenance hazard: Smokey Remains

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

tboi unlock

This implements a new rare maintenance spawn, smokey remains. Piles of bones that used to be Nanotrasen employees, still packed with whatever reagent finally did them in.

Rarely replacing a crate spawn (7/1007 chance per crate spawn if I'm doing the math right), these act as landmines for uncareful travelers in maintenance. When you get too close, one will release a cloud of a randomly generated reagents into the air (the same reagent every time, unique to the instance of remains). You can (mostly) avoid this by walking instead of running. These will "re-arm" themselves after 4-6 minutes. They're decals, so you can just sweep them away if they're a problem.

This also slightly tweaks the original iteration of this, which I used in my Smoking Room ruin. Don't worry about it.

I'm open to adjusting the rarity, cooldown, and selection of reagents if need be.
## Why It's Good For The Game

Makes maintenance a bit more spicy. Maybe dangerous, maybe beneficial, maybe it's baldium. Who knows! Better be careful just in case!

There's plenty of boring chemicals that could roll, but an interesting one could be a nice injection of uniqueness into an otherwise dull round. It's an infinite supply, but can only be dispensed infrequently (at a fixed location) and would be extremely difficult to harvest.

The rarity, combined with the variety of reagents that can be picked, make it unlikely that these will have any serious impact on a round (it's still possible, but rarely, which is good).
## Changelog
:cl: Rhials
add: Smokey remains have appeared in maintenance. Make sure to walk when near them!
/:cl:

### CODE REVIEW

**Comment by SomeoneYouProbablyKnow** [comment]
Neat. Make sure you can't turn those into chem factories for some fucked up chemicals, or say that's an intended use case.

**Comment by Time-Green** [comment]
> Neat. Make sure you can't turn those into chem factories for some fucked up chemicals, or say that's an intended use case.

Only chems in the stomach can be "recovered". The only other way is through gibbing and extracting it from the meat, which I kinda respect

**Comment by Rhials** [comment]
> Neat. Make sure you can't turn those into chem factories for some fucked up chemicals, or say that's an intended use case.

It's intended, since if someone wants to go through all that effort to squeeze out a few units every couple of minutes, I'd say that's the sandbox being well-leveraged. That's also the reason for the delay.

**Comment by EricZilla** [comment]
Spooky

**Comment by norsvenska** [comment]
chemical warfare: maintenance pill edition

**Comment by Dawnseer** [comment]
I am looking forward to stepping on one and getting dozed with adminordrazine

**Comment by Time-Green** [comment]
> I am looking forward to stepping on one and getting dozed with adminordrazine

At no point in this game have random reagents ever allowed adminordrazine

**Comment by Terra1** [comment]
Romerol?

File: code/game/objects/effects/decals/remains.dm [hunk 1]
```diff
@@ -49,6 +54,28 @@
...
+	var/static/list/random_reagents = list()
+	if(!length(random_reagents))
+		for(var/datum/reagent/reagent_path as anything in subtypesof(/datum/reagent))
+			if(!(initial(reagent_path.chemical_flags) & REAGENT_CAN_BE_SYNTHESIZED))
+				continue
+			if(istype(reagent_path, /datum/reagent/medicine) || istype(reagent_path, /datum/reagent/consumable)) //Boooooriiiiing
+				continue
+			random_reagents += reagent_path
+	var/picked_reagent = pick(random_reagents)
+	return picked_reagent
```
Comment (Time-Green) [inline code/game/objects/effects/decals/remains.dm:68]: Please use get_random_reagent_id() instead. You can make it respect a blacklist to get the same effect (or flags or something, but blacklist is probably easiest)

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Comment by Dawnseer** [comment]
> At no point in this game have random reagents ever allowed adminordrazine

Just crush my dreams why don't you

**Comment by DrAmazing343** [comment]
this that shit that killed john nanotrasen

File: code/game/objects/effects/decals/remains.dm [hunk 2]
```diff
@@ -18,29 +18,38 @@
...
 	. = ..()
 
-	proximity_monitor = new(src, 0)
-	that_shit_that_killed_saddam = get_random_reagent_id()
+	proximity_monitor = new(src, 1)
+	var/list/blocked_reagents = list()
+	for(var/datum/reagent/reagent_path as anything in subtypesof(/datum/reagent))
+		if(ispath(reagent_path, /datum/reagent/medicine) || ispath(reagent_path, /datum/reagent/consumable)) //Boooooriiiiing
+			blocked_reagents += reagent_path
+	that_shit_that_killed_saddam = get_random_reagent_id(blacklist = blocked_reagents)
```
Comment (Time-Green) [inline code/game/objects/effects/decals/remains.dm:40]: You can just use subtypesof on the consumable and medicine typepaths
```suggestion
	var/list/blacklisted_reagents = subtypesof(/datum/reagent/medicine) + subtypesof(/datum/reagent/consumable)
	that_shit_that_killed_saddam = get_random_reagent_id(blacklist = blacklisted_reagents)
```
(you can also put it directly in the proc I don't really care)

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Pull request merged**